metricflow/specs/python_object.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from typing import List
+
+from dbt_semantic_interfaces.call_parameter_sets import ParseWhereFilterException
+from dbt_semantic_interfaces.implementations.filters.where_filter import PydanticWhereFilter
+
+from metricflow.naming.linkable_spec_name import StructuredLinkableSpecName
+from metricflow.protocols.query_parameter import GroupByParameter
+from metricflow.query.query_exceptions import InvalidQueryException
+from metricflow.specs.query_param_implementations import DimensionOrEntityParameter, TimeDimensionParameter
+
+
+def parse_object_builder_naming_scheme(group_by_item_name: str) -> GroupByParameter:
+    """Parses a string following the object-builder naming scheme into the corresponding GroupByParameter.
+
+    The implementation of the query parameter classes seems incomplete and there needs to be follow up with the author
+    of the query interface classes for the best approach. Right now, it seems like using the where filter is the only
+    way to handle this conversion. However, it seems like this functionality should be abstracted into a module that
+    handles operations related to the object-builder naming scheme.
+
+    Additional issues:
+    * The call parameter sets in DSI does not support date part.
```

Comment:
Let's also not support date part in saved queries, it's a partially supported feature built solely for use in interactive sessions through the Tableau connector. As we get more experience with how the date_part mechanics are used in practice we'll push more robust support down to MetricFlow.